src/calc-length.js

```diff
@@ -93,6 +93,72 @@
     return new CalcLength(calcDictionary);
   };
 
+  CalcLength.prototype._addCalcLengths = function(addedLength) {
+    if (!(addedLength instanceof CalcLength)) {
+      throw new TypeError('Objects not of type CalcLength');
+    } 
+    
+    var calcDictionary = {};
+
+    //Iterate through all possible length types and add there values
+    for(var i = 0; i < shared.LengthValue.LengthType.length; i++){
+      var type = shared.LengthValue.LengthType[i];
+      if (this[type] == null){
+        calcDictionary[type] = addedLength[type];
+      } else if (addedLength[type] == null) {
+        calcDictionary[type] = this[type];
+      } else {
+        calcDictionary[type] = this[type] + addedLength[type];
+      }
+    }
+
+    return new CalcLength(calcDictionary);
+  };
+
+  CalcLength.prototype._subtractCalcLengths = function(subtractedLength) {
+    if (!(subtractedLength instanceof CalcLength)) {
+      throw new TypeError('Objects not of type CalcLength');
+    } 
+    
+    var calcDictionary = {};
+
+    //Iterate through all possible length types and add there values
+    for(var i = 0; i < shared.LengthValue.LengthType.length; i++){
+      var type = shared.LengthValue.LengthType[i];
+      if (this[type] == null && subtractedLength[type] == null) {
+        calcDictionary[type] = null;
```

Comment:
In the constructor, you check if things are undefined, so it's fine to skip this case and end up with calcDictionary[type] being undefined.